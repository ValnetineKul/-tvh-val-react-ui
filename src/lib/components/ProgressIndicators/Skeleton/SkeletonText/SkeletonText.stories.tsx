import { createMeta, createTemplate } from '../../../../story-utils';
import SkeletonText from './SkeletonText';

const Template = createTemplate(SkeletonText);

export const Primitive = Template.bind({});
Primitive.args = {
  heading: 'h1',
  body: 'body1',
  bodyLinesCount: 3,
  width: '100%',
};
Primitive.storyName = 'SkeletonText';

export default createMeta({
  component: SkeletonText,
  title: 'Components/ProgressIndicators/Skeleton/SkeletonText',
  argTypes: {
    headingClassName: {
      control: {
        disable: true,
      },
    },
    bodyTextClassName: {
      control: {
        disable: true,
      },
    },
    lastLineClassName: {
      control: {
        disable: true,
      },
    },
  },
});
