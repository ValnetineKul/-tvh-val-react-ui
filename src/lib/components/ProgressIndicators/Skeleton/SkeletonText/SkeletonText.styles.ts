import type { Surface } from '../../../../../themes/core';
import { makeStyles } from '../../../../../themes/core';
import { skeletonAnimationStyles } from '../helper';

type Props = {
  onSurface: Surface;
  width?: number | string;
};

const useStyles = makeStyles<Props, 'h1' | 'h2' | 'h3' | 'h4' | 'body1' | 'body2'>({
  name: 'SkeletonText',
})((theme, { onSurface, width }) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    width,
  },
  skeletonAnimation: {
    ...skeletonAnimationStyles(
      onSurface === '200' ? theme.color['bg/surface300/default'] : theme.color['bg/surface200/default'],
      onSurface === '200' ? theme.color['bg/surface400/default'] : theme.color['bg/surface150/default']
    ),
  },
  h1: {
    height: theme.font['screen/heading1/functional/xs'].fontSize,
    marginBottom: '6px',
    [theme.breakpoints.up('sm')]: {
      height: theme.font['screen/heading1/functional/sm'].fontSize,
      marginBottom: '5px',
    },
  },
  h2: {
    height: theme.font['screen/heading2/functional/xs'].fontSize,
    marginBottom: '6px',
    [theme.breakpoints.up('sm')]: {
      height: theme.font['screen/heading2/functional/sm'].fontSize,
      marginBottom: theme.layoutSpacing['spacing/200'],
    },
  },
  h3: {
    height: theme.font['screen/heading3/functional/xs'].fontSize,
    marginBottom: '3px',
    [theme.breakpoints.up('sm')]: {
      height: theme.font['screen/heading3/functional/sm'].fontSize,
      marginBottom: '6px',
    },
  },
  h4: {
    height: theme.font['screen/heading4/functional/xs'].fontSize,
    marginBottom: '6px',
    [theme.breakpoints.up('sm')]: {
      height: theme.font['screen/heading4/functional/sm'].fontSize,
      marginBottom: theme.layoutSpacing['spacing/200'],
    },
  },
  heading: {
    width: '198px',
  },
  bodyWrapper: {
    display: 'flex',
    flexDirection: 'column',
    marginTop: theme.layoutSpacing['spacing/300'],
  },
  body1: {
    height: theme.layoutSpacing['spacing/400'],
    marginTop: theme.layoutSpacing['spacing/200'],
    marginBottom: theme.layoutSpacing['spacing/200'],
    ':last-child': {
      width: '60%',
    },
  },
  body2: {
    height: theme.layoutSpacing['spacing/350'],
    marginTop: theme.layoutSpacing['spacing/100'],
    marginBottom: theme.layoutSpacing['spacing/100'],
    ':last-child': {
      width: '60%',
    },
  },
}));

export default useStyles;
