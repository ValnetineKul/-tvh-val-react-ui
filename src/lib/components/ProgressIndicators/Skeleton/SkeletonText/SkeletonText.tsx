import type { FC } from 'react';
import React from 'react';
import { Skeleton } from '@mui/material';
import { useSurface } from '../../../../../themes/core';
import useStyles from './SkeletonText.styles';

export interface SkeletonTextProps {
  heading?: 'h1' | 'h2' | 'h3' | 'h4';
  body?: 'body1' | 'body2';
  bodyLinesCount: number;
  width?: number | string;
  containerClassName?: string;
  headingClassName?: string;
  bodyTextClassName?: string;
  lastLineClassName?: string;
}

const SkeletonText: FC<SkeletonTextProps> = ({
  heading,
  body = 'body1',
  bodyLinesCount = 3,
  width = '100%',
  headingClassName,
  bodyTextClassName,
  lastLineClassName,
  containerClassName,
}) => {
  const { color: onSurface } = useSurface();
  const { classes, cx } = useStyles({ onSurface, width });

  return (
    <div className={cx(classes.root, containerClassName)} data-testid="skeleton-text">
      {heading && (
        <Skeleton
          component="div"
          variant="rectangular"
          animation="wave"
          className={cx(classes[heading], classes.skeletonAnimation, classes.heading, headingClassName)}
        />
      )}

      {bodyLinesCount > 0 && (
        <div className={classes.bodyWrapper}>
          {new Array(bodyLinesCount).fill(null).map((_, i) => (
            <Skeleton
              component="div"
              variant="rectangular"
              animation="wave"
              key={i}
              className={cx(
                classes[body],
                classes.skeletonAnimation,
                bodyTextClassName,
                bodyLinesCount - 1 === i ? lastLineClassName : null
              )}
            />
          ))}
        </div>
      )}
    </div>
  );
};

export default SkeletonText;
