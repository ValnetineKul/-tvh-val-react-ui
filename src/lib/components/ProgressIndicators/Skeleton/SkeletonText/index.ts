import SkeletonText from './SkeletonText';

export type { SkeletonTextProps } from './SkeletonText';
export default SkeletonText;
