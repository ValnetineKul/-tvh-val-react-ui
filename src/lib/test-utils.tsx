import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
