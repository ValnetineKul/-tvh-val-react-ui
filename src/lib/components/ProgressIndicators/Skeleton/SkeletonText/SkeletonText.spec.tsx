import React from 'react';
import SkeletonText from './SkeletonText';
import { render } from '../../../../test-utils';

describe('SkeletonText', () => {
  it('Should render correctly', () => {
    const { container } = render(<SkeletonText bodyLinesCount={3} heading="h1" />);
    expect(container.firstChild).toMatchSnapshot('SkeletonText');
  });
});
