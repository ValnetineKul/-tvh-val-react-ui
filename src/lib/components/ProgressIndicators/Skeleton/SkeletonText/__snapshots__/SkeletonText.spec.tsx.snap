// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonText Should render correctly: SkeletonText 1`] = `
<div
  class="tss-lplq1l-SkeletonText-root"
  data-testid="skeleton-text"
>
  <div
    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-lg6l74-SkeletonText-h1-SkeletonText-skeletonAnimation-SkeletonText-heading mui-1slz293-MuiSkeleton-root"
  />
  <div
    class="tss-z0i8lo-SkeletonText-bodyWrapper"
  >
    <div
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-yvp8fo-SkeletonText-body1-SkeletonText-skeletonAnimation mui-1slz293-MuiSkeleton-root"
    />
    <div
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-yvp8fo-SkeletonText-body1-SkeletonText-skeletonAnimation mui-1slz293-MuiSkeleton-root"
    />
    <div
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-yvp8fo-SkeletonText-body1-SkeletonText-skeletonAnimation mui-1slz293-MuiSkeleton-root"
    />
  </div>
</div>
`;
